File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/Inventory/StockingDetails.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.Inventory
{
    public class StockingDetails : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public long CurrentStock { get; set; }
        public long ActualStock { get; set; }
        public long TheDifference { get; set; }

        [ForeignKey("ShopItem")]
        public long ShopItemID { get; set; }

        [ForeignKey("Stocking")]
        public long StockingId { get; set; }

        public ShopItem ShopItem { get; set; }
        public Stocking Stocking { get; set; }

    }
}

File: LMS_CMS_Frontend/src/app/Models/Inventory/stocking-details.ts
export class StockingDetails {
    constructor(
        public id: number = 0,
        public currentStock:number =0,
        public actualStock :number|any =0,
        public theDifference :number|any =0,
        public shopItemID :number =0,
        public stockingId :number =0,
        public shopItemName :string = '',
        public barCode: string = '',
        public insertedAt :string ="",
        public ItemPrice : number =0 ,
        public insertedByUserId :number =0,
    ) {}
}
